src/didactUriCompletionItemProvider.ts

```diff
@@ -0,0 +1,314 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as extensionFunctions from './extensionFunctions';
+import {DidactUri} from './didactUri';
+
+let extContext: vscode.ExtensionContext;
+
+export function setContext(ctxt: vscode.ExtensionContext) {
+	extContext = ctxt;
+}
+
+export class DidactUriCompletionItemProvider implements vscode.CompletionItemProvider {
+	
+	provideCompletionItems(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken, context: vscode.CompletionContext): vscode.ProviderResult<vscode.CompletionItem[] | vscode.CompletionList> {
+		let lineToSearch = document.lineAt(position).text;
+		if (lineToSearch.indexOf('didact://?') > -1) {
+			return this.provideCompletionItemsForDidactProtocol(lineToSearch);
+		}
+		return this.provideCompletionItemsOutsideDidactURI(lineToSearch);
+	}
+
+	public provideCompletionItemsOutsideDidactURI(lineToSearch: string) : vscode.CompletionItem[] {
+		let completions: vscode.CompletionItem[] = [];
+
+		if (lineToSearch.indexOf('didact://?') === -1) {
+			// Add Validate All button
+			this.insertValidateAllButtonCompletion("Insert Validate All Button", completions);
+
+			// Add requirements label
+			this.insertNamedStatusLabelForRequirementCompletion("Insert Requirements Label", completions);
+
+			// Add link to install required VS Code extension
+			this.insertInstallExtensionLinkCompletion("Insert link to install required VS Code extension", completions);
+
+			// Add link to create temp folder and set it as the workspace root
+			this.insertAddWorkspaceFolderLinkCompletion("Insert link to create a temporary folder as WS root", completions);
+
+			// Add link to open didact file from elsewhere in the extension 
+			this.insertRedhatDidactLinkCompletion("Insert link to start Didact from File Elsewhere in Extension Folder", completions);
+
+			// Add didact starter 
+			this.insertDidactProtocolStarterCompletion("Start a didact link", completions);
+		}
+
+		return completions;
+	}
+
+	public provideCompletionItemsForDidactProtocol(text: string): vscode.CompletionItem[] {
+		let completions: vscode.CompletionItem[] = [];
+		if (text.indexOf('didact://?') > -1) {
+			const start = text.indexOf('didact://?');
+			const end = text.indexOf(' ', start);
```

Comment:
Was trying to figure out a good way of identifying the end of the Didact link and space was the first thing that came up. It can also be a ')' as the end of a `[some text](some link)` setup. Wondering if maybe we should use a regex, but I didn't come up with a good one that made sense. Any ideas @apupier ?

Reply:
Maybe something like this -- https://www.regextester.com/94092

Reply:
just not sure how to make that work in typescript...

Reply:
https://github.com/jboss-fuse/vscode-datavirt/blob/master/src/utils.ts#L91-L97

Reply:
Regex works now. Huzzah